R/derive_param_confirmed_bor.R

```diff
@@ -0,0 +1,528 @@
+#'   deriving the new parameter are restricted to observations up to the
+#'   specified date. Observations at the specified date are included. For
+#'   subjects without first PD date all observations are take into account.
+#'
+#'   *Permitted Values:* a `date_source` object (see `admiral::date_source()`
+#'   for details)
+#'
+#'   *Default:* `NULL`,
+#'
+#' @param source_datasets Source dataset for the first PD date
+#'
+#'   A named list of datasets is expected. It links the `dataset_name` from
+#'   `source_pd` with an existing dataset.
+#'
+#'   For example if `source_pd = pd_date` with
+#'   ```
+#'   pd_date <- date_source(
+#'     dataset_name = "adrs",
+#'     date = ADT,
+#'     filter = PARAMCD == PD
+#'   )
+#'   ```
+#'   and the actual response dataset in the script is `myadrs`, `source_datasets
+#'   = list(adrs = myadrs)` should be specified.
+#'
+#'   This allows to define `pd_date` at a higher level, e.g., at company level,
+#'   where the actual dataset does not exist.
+#'
+#' @param reference_date Reference date
+#'
+#'   The reference date is used for the derivation of `"SD"` and
+#'   `"NON-CR/NON-PD"` response (see "Details" section). Usually it is treatment
+#'   start date (`TRTSDT`) or randomization date (`RANDDT`).
+#'
+#'   *Permitted Values:* a numeric date variable
+#'
+#' @param ref_start_window Stable disease time window
+#'
+#'   Assessments at least the specified number of days after the reference date
+#'   with response `"CR"`, `"PR"`, `"SD"`, or `"NON-CR/NON-PD"` are considered
+#'   as `"SD"` or `"NON-CR/NON-PD"` response.
+#'
+#'   *Permitted Values:* a non-negative numeric scalar
+#'
+#' @param ref_confirm Minimum time period for confirmation
+#'
+#'   The assessment and the confirmatory assessment for `"CR"` and `"PR"` have
+#'   to be at least the specified number of days apart.
+#'
+#' @param max_nr_ne
+#'
+#'   The specified number of `"NE"` assessments between the assessment and the
+#'   confirmatory assessment for `"CR"` and `"PR"` response is accepted.
+#'
+#'   *Permitted Values:* a non-negative numeric scalar
+#'
+#'   *Default:* `1`
+#' @param accept_sd Accept `"SD"` for `"PR"`?
+#'
+#'   If the argument is set to `TRUE`, one `"SD"` assessment between the
+#'   assessment and the confirmatory assessment for `"PR"` response is accepted.
+#'   Otherwise, no `"SD"` assessment must occur between the two assessments.
+#'
+#'   *Permitted Values:* a logical scalar
+#'
+#'   *Default:* `FALSE`
+#'
+#' @param missing_as_ne Consider no assessments as `"NE"`?
+#'
+#'   If the argument is set to `TRUE`, the response is set to `"NE"` for
+#'   subjects without an assessment in the input dataset. Otherwise, the
+#'   response is set to `"MISSING"` for these subjects.
+#'
+#'   *Permitted Values:* a logical scalar
+#'
+#'   *Default:* `FALSE`
+#'
+#' @param aval_fun Function to map character analysis value (`AVALC`) to numeric
+#'   analysis value (`AVAL`)
```

Comment:
In https://github.com/pharmaverse/admiralonco/issues/19 we decided we need this as the `AVAL` values are company specific. We can not deduce them from the input dataset because it may not contain all values. For example, the input dataset may not contain any observation with `AVALC == "NE"` but observations with `AVALC == "NE"` may be added to the output dataset for patients without observations in the input dataset.

Reply:
ah i see your point, but my thinking is would users really care if we just assigned NE always as say AVAL=6? i'm just wondering how often users are really going to need to update this vs the added complexity of another argument. but maybe this sets us up for more flexibility that the users really require.

Reply:
What is the benefit of removing the argument?

The parameter has a default value. If the users are happy with the default value, they do not need to specify the argument. If they need something else, it is clear how the derivation of `AVAL` can be changed. Without the argument they would need to add a post-processing step which overwrites the `AVAL` values derived by the function. I would avoid such post-processing steps because they make the code less readable and harder to review.

Reply:
the more arguments we have the more complex it is for users to understand, so i always try to cut down any i dont think necessary. if you feel strongly this adds value then go ahead and leave in and we can always see if testers give any feedback. as you say though most will probably never change default and never even need to notice it

Reply:
I do not think that "the more arguments we have the more complex it is for users to understand" holds in general. I think removing `aval_fun` would not reduce the complexity of the documentation or code. Thus I would keep it.

Reply:
the keyword would have been: _the more `unnecessary` arguments we have the more complex it is for users to understand_, but happy to go with your call here and let's leave it. i spotted we used similar on an admiral core function too so we have somewhat of a convention here we can follow.